tests/decide_gi.py

```diff
@@ -0,0 +1,101 @@
+
+expected_results = {}
+expected_results['colorref_smallexample_2_49'] = {0:
+                                                      {1: (True, True)}
+                                                  }
+expected_results['colorref_smallexample_4_7'] = {0:
+                                                      {1: (False, False), 2: (True, False), 3: (False, False)},
+                                                 1:
+                                                     {2: (False, False), 3: (True, True)},
+                                                 2:
+                                                     {3: (False, False)}
+
+                                                 }
+expected_results['colorref_smallexample_6_15'] = {0:
+                                                      {1: (True, True), 2: (False, False), 3: (False, False), 4: (False, False), 5: (False, False)},
+                                                  1:
+                                                      {2: (False, False), 3: (False, False), 4: (False, False), 5: (False, False)},
+                                                  2:
+                                                      {3: (True, True), 4: (False, False), 5: (False, False)},
+                                                  3:
+                                                      {4: (False, False), 5: (False, False)},
+                                                  4:
+                                                      {5: (True, False)},
+                                                 }
+
+def test_balanced_or_bijected(graph: 'Graph', is_balanced: bool, is_bijected: bool):
+    """
+    Tests the is_balanced_or_bijected method of decide_gi.py agains expected answers for is_balanced and is_bijected
+    :param graph: The graph to test
+    :param is_balanced: the expected answer for is_balanced
+    :param is_bijected: the expected answer for is_bijected
+    :return: bool test passes == True, test fails == False
+    """
+    test_is_balanced, test_is_bijected = is_balanced_or_bijected(graph)
+
+    if not (test_is_balanced == is_balanced and test_is_bijected == is_bijected):
+        print('DEBUG...')
+        print('Should be  : {}, {}'.format(is_balanced, is_bijected))
+        print('Calculated : {}, {}'.format(test_is_balanced, test_is_bijected))
+    return test_is_balanced == is_balanced and test_is_bijected == is_bijected
+
+
+# Testing loop
+# A disjoint union is coloured using color_refinement(degree_color_initialization(graph))
+# The method is_balanced_or_bijected is tested against a hardcoded dictionary containing the expected outcomes.
+#
+# Currently the following graph files are tested:
+#     - colorref_smallexample_2_49
+#     - colorref_smallexample_4_7
+#     - colorref_smallexample_6_15
+
+error_count = 0
+graph_count = 0
+for file in files:
+    filename = '../test_graphs/color_refinement/' + file + '.grl'
+    with open(filename) as f:
```

Comment:
In input_output/file_output (wat file_io moet heten besef ik me nu), zit een methode load_graph_list, waarin dit is geimplementeerd. Dan krijg je ook gelijk de L[0], zodat je dat hier nergens meer hoeft te doen.